File: 2022/cpp/day1.cpp
#include <vector>
#include <iostream>
#include <string>
#include <numeric>
#include <fstream>
using namespace std;

int main() {
    vector<int> elf;
    ifstream infile;
    infile.open("input/day1.txt", ios::in);
    string cur;
    int c = 0;
    while (getline(infile, cur)) {
        bool digit = false;
        if (cur[0] >= '0' && cur[0] <= '9') digit = true;
        if (digit) c += stoi(cur);
        else {
            elf.push_back(c);
            c = 0;
        }
    }
    sort(elf.begin(), elf.end());
    cout << elf[elf.size() - 1] << '\n';
    cout << accumulate(elf.end()-3, elf.end(), 0) << '\n';
    return 0;
}

File: 2022/py/day1.py
with open("input/day1.txt", "r") as f:
    data = f.read().split("\n\n")
    s = []
    for elf in data:
        cur = 0
        for c in elf.splitlines():
            cur += c
        s.append(cur)
    s= sorted(s)

print(s[-1])
print(sum(s[-3:]))